src/nodes_gpu/stats.rs

```diff
@@ -0,0 +1,160 @@
+        let shader = compute_shader::load(device.clone()).unwrap();
+        let pipeline = ComputePipeline::new(
+            device.clone(),
+            shader.entry_point("main").unwrap(),
+            &(),
+            None,
+            |_| {},
+        )
+        .unwrap();
+
+        Ok(Stats { device, pipeline, queue, input: parameters.take("input")? })
+    }
+}
+
+#[async_trait]
+impl ProcessingNode for Stats {
+    async fn pull(
+        &self,
+        frame_number: u64,
+        _puller_id: NodeID,
+        context: &ProcessingContext,
+    ) -> Result<Payload> {
+        let input = self.input.pull(frame_number, context).await?;
+
+        let (frame, fut) =
+            ensure_gpu_buffer::<Raw>(&input, self.queue.clone()).context("Wrong input format")?;
+
+        let sink_buffer = DeviceLocalBuffer::<[u8]>::array(
+            self.device.clone(),
+            (1 << frame.interp.bit_depth) * 4, // actually uint
+            BufferUsage {
+                storage_buffer: true,
+                storage_texel_buffer: true,
+                transfer_src: true,
+                transfer_dst: true,
+                vertex_buffer: true,
+                ..BufferUsage::none()
+            },
+            std::iter::once(self.queue.family()),
+        )?;
+
+        let push_constants = compute_shader::ty::PushConstantData {
+            width: frame.interp.width as _,
+            height: frame.interp.height as _,
+        };
+
+        let layout = self.pipeline.layout().set_layouts()[0].clone();
+        let set = PersistentDescriptorSet::new(
+            layout,
+            [
+                WriteDescriptorSet::buffer(0, frame.storage.untyped()),
+                WriteDescriptorSet::buffer(1, sink_buffer.clone()),
+            ],
+        )
+        .unwrap();
+
+        let mut builder = AutoCommandBufferBuilder::primary(
+            self.device.clone(),
+            self.queue.family(),
+            OneTimeSubmit,
+        )
+        .unwrap();
+        builder
+            .fill_buffer(FillBufferInfo {
+                data: 0,
+                ..FillBufferInfo::dst_buffer(sink_buffer.clone())
+            })?
+            .bind_descriptor_sets(
+                PipelineBindPoint::Compute,
+                self.pipeline.layout().clone(),
+                0,
+                set,
+            )
+            .push_constants(self.pipeline.layout().clone(), 0, push_constants)
+            .bind_pipeline_compute(self.pipeline.clone())
+            .dispatch([
+                (frame.interp.width as u32 + 15) / 16,
+                (frame.interp.height as u32 + 31) / 32,
+                1,
+            ])?;
+        let command_buffer = builder.build()?;
+
+        let future =
+            fut.then_execute(self.queue.clone(), command_buffer)?.then_signal_fence_and_flush()?;
+
+        future.wait(None).unwrap();
+        Ok(Payload::from(Frame {
+            interp: Raw {
+                width: 4096,
+                height: 1,
```

Comment:
I am a bit unsure if this is the right format to do this.
Also cant we name the node hist or something more specific?

Reply:
What format would you use?

Reply:
I would use something more explicit. Like a Stats type

Reply:
Hmm yes that could work, but then you could no longer use the `RawBlobWriter` for example to write these to disk. 
Maybe this is a more general problem that needs a "proper" solution and not just this "hack".